### ISSUE TITLE
Truffle console command throws error trying to read json files built from Vyper source files.

### ISSUE THREAD

**Issue by dPreininger** [trufflesuite/truffle#5812]
- [ x] I've [opened a support ticket](https://trufflesuite.zendesk.com/hc/en-us/requests/new) before filing this issue.

---------------------------

## Issue

Truffle  console command crashes on .json files generated from Vyper source files. Not all versions of Truffle/Vyper compiler seem to be affected.

## Steps to Reproduce

Be sure you are using Ubuntu based OS with Bash terminal.
Clone this repository: https://github.com/dPreininger/vyper-truffle-report
Run `bash run.sh`. This will run included build files and return an error (read README for more details).

## Expected Behavior

Truffle console command should not exit.

## Actual Results

Truffle console exits with this error message:
```console
Unexpected error setting up the environment or provisioning contracts while instantiating the console.
Error: Error parsing or reading VyperStorage.json: Unexpected token u in JSON at position 0
    at {{ MY_HOME_DIR }}/.nvm/versions/node/v18.12.1/lib/node_modules/truffle/build/webpack:/packages/core/lib/console.js:283:1
    at Array.forEach (<anonymous>)
    at Console.provision ({{ MY_HOME_DIR }}/.nvm/versions/node/v18.12.1/lib/node_modules/truffle/build/webpack:/packages/core/lib/console.js:256:1)
    at Console.start ({{ MY_HOME_DIR }}/.nvm/versions/node/v18.12.1/lib/node_modules/truffle/build/webpack:/packages/core/lib/console.js:130:1)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at Object.module.exports [as run] ({{ MY_HOME_DIR }}/.nvm/versions/node/v18.12.1/lib/node_modules/truffle/build/webpack:/packages/core/lib/commands/console/run.js:32:1)
    at runCommand ({{ MY_HOME_DIR }}/.nvm/versions/node/v18.12.1/lib/node_modules/truffle/build/webpack:/packages/core/lib/command-utils.js:201:1)
```
The error is caused by console.js file (https://github.com/trufflesuite/truffle/blob/develop/packages/core/lib/console.js), line 266.
The code expects the VyperStorage.json file to include metadata key. However, Vyper/Truffle does not generate this data, hence we are calling JSON.parse(undefined), which throws an error.

I am willing to contribute to this repository to fix this issue, however I am unsure what exactly is the correct behavior here. It seems to me there are two possibilities:

1. Metadata field is optional, there is a bug because the field is treated as required in the code.
2. Metadata field is required, there is a bug because the field is only generated from Solidity source files and from Vyper files.

Clearing this up would enable me to fix the bug, if the maintainers agree.

## Environment

* Operating System: Ubuntu 22.04 LTS
* Ethereum client: Ganache v7.6.0
* Truffle version: Truffle v5.7.1 (core: 5.7.1)
* node version: Tested with 16.15.1 and 18.12.1
* npm version: 8.19.2
* vyper compiler: 0.3.7+commit.6020b8b

**Comment by cds-amal** [comment]
Thanks for raising this issue/bug @dPreininger! Apologies for the delayed response. We're only now returning from holiday. 

Truffle should not crash if metadata is undefined; that filter logic should be guarded with something like:

```javascript
if (json.metadata !== undefined) {
  const metadata = JSON.parse(json.metadata);
  // filter logic
}
```
If you have the time to make the PR, please let us know, otherwise, one of us will grab this one. I'd like to see it get into the next release.

**Comment by dPreininger** [comment]
I've opened the PR [#5819](https://github.com/trufflesuite/truffle/pull/5819)

**Comment by cds-amal** [comment]
Hi @dPreininger, this but still exists for the current Truffle version. In this case, since there's a PR to address this bug, our policy is to close it when the next Truffle version is released. This fix should make this week's release. Thanks so much for contributing :bow:

**Comment by lsqproduction** [comment]
Yay. Fix and releases in [Truffle v5.7.2](https://github.com/trufflesuite/truffle/releases/tag/v5.7.2)
